flycheck-rust.el

```diff
@@ -145,12 +145,25 @@ description of the conventional Cargo project layout."
                       (file-equal-p dir (file-name-directory target-path))))
                   ;; build a list of (target . dir) candidates
                   (-table-flat
-                   'cons targets
+                   'cons (-flatten-n 1 targets)
                    (flycheck-rust-dirs-list file-name
                                             (file-name-directory manifest)))))
             ;; If all else fails, just pick the first target
-            (car targets))))
-      (let-alist target (cons (flycheck-rust-normalize-target-kind .kind) .name)))))
+            (car (car targets)))))
+      (let-alist target
+        ;; If target is 'custom-build', we search other target
+        (if (string= "custom-build" (car .kind))
+            (--> targets
+                 ;; filtering same packages as current buffer
+                 (--filter
+                  (--any?
+                   (let-alist it
+                     (string= file-name .src_path)) it) it)
+                 (--remove
+                  (let-alist it
+                    (string= "custom-build" (car .kind))) (car it))
+                 (let-alist (car it) (cons (car .kind) .name)))
+          (cons (flycheck-rust-normalize-target-kind .kind) .name))))))
```

Comment:
I think this can be simplified to:

```emacs-lisp
      ;; If target is 'custom-build', we search other target
      (when (string= "custom-build" (let-alist target (car .kind)))
        (setq target (--> packages
                          ;; filtering same packages as current buffer
                          (--find (--any? (equal target it) it) it)
                          (--find (not (equal target it)) it))))
      (when target
        (let-alist target
          (cons (flycheck-rust-normalize-target-kind .kind) .name))))))
```

This uses `equal` instead of `string=` comparisons to find the right package and target, and it makes sure we normalize the target name in both cases.  It also returns `nil` when there are no other targets in the package containing the `custom-build` package, should that case arrive.